### DISCUSSION TITLE
Sample project for Rust is non functional on replit

### DISCUSSION THREAD

**Discussion by Larandar** [BattlesnakeOfficial/feedback#187] [Bug Reports]
Step to reproduce:

- Open https://docs.battlesnake.com/guides/getting-started
- Select "Open project in Replit"
- Try to push the "Run" button

Possible fix:

- Be clear that the Rust getting is not a functional on Replit and require self hosting

**Comment by Nettogrof** [comment]
And just running the command 'cargo build' in the shell  result in errors : 
`error: failed to run custom build command for `rocket v0.4.10`

Caused by:
  process didn't exit successfully: `/home/runner/starter-snake-rust/target/debug/build/rocket-3d836f63d287ffc5/build-script-build` (exit code: 101)
--- stderr
Error: Rocket (core) requires a 'dev' or 'nightly' version of rustc.
Installed version: 1.44.0 (2020-06-01)
Minimum required:  1.54.0-nightly (2021-05-18)
See the getting started guide (https://rocket.rs/v0.5/guide/getting-started/) for more information.
thread 'main' panicked at 'Aborting compilation due to incompatible compiler.', /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket-0.4.10/build.rs:29:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

warning: build failed, waiting for other jobs to finish...
error: failed to run custom build command for `rocket_codegen v0.4.10`

Caused by:
  process didn't exit successfully: `/home/runner/starter-snake-rust/target/debug/build/rocket_codegen-854020e3f70db9b6/build-script-build` (exit code: 101)
--- stderr
Error: Rocket (codegen) requires a 'dev' or 'nightly' version of rustc.
Installed version: 1.44.0 (2020-06-01)
Minimum required:  1.33.0-nightly (2019-01-13)
See the getting started guide (https://rocket.rs/v0.5/guide/getting-started/) for more information.
thread 'main' panicked at 'Aborting compilation due to incompatible compiler.', /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket_codegen-0.4.10/build.rs:29:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

warning: build failed, waiting for other jobs to finish...
error: build failed`

**Comment by coreyja** [comment]
Oh I think part of the issue is that `rocket` 0.4 requires nightly Rust and that either isn't supported on Replit, or isn't built into the Run it button (I haven't looked)

The newer version of rocket `0.5-rc` runs on Stable which I think would make the starter easier to work with. I might try to get the starter updated over the weekend so it runs on Stable!

**Reply by coreyja** [reply]
Opened a PR to upgrade to the latest Rocket and a few other changes to get it working on Replit!
https://github.com/BattlesnakeOfficial/starter-snake-rust/pull/4

@Larandar if you wanted to use it on Replit you can give this link a shot, while that PR is still open
[replit.com/github/coreyja/starter-snake-rust](https://replit.com/github/coreyja/starter-snake-rust)

**Comment by Larandar** [comment]
I don't think running on stable is the solution, TBH unless I do production
code I use nightly everywhere. I think the underlying issue is that replit
install Rust without using rustup which makes changing channels impossible.

I tried to fight with replit yesterday, using the new "Nix (Beta)" template
but I have to say I'm not impressed with replit usability when it comes to
"just restart the damn container".

**Reply by coreyja** [reply]
> I think the underlying issue is that replit install Rust without using rustup which makes changing channels impossible

> using the new "Nix (Beta)" template

Yup I think we ended up at basically the same spot!

After looking at it more you are right, I don't think nightly was 'directly' related. But seemed like an annoying this to require for the starter snake 🤷 

> I have to say I'm not impressed with replit usability when it comes to "just restart the damn container".

I will have to agree with you there lol. Replit wasn't a great experience for Rust I felt like

**Comment by Larandar** [comment]
I think I have a working prototype, will clean that up and then do a PR

Unrelated issue, but is there a type definition for battlesname that can be
used for generating the types? I found
https://docs.battlesnake.com/references/api but I would like a more
"parsing friendly" version :)
